### ISSUE TITLE
"this & object prototypes": cover `__proto__` in object literals

### ISSUE THREAD

**Issue by getify** [getify/You-Dont-Know-JS#323]
ES6 now allows:

``` js
var Foo = { .. };

var Bar = {
   __proto__: Foo,
   ..
};
```

This is obviously now much preferable in OLOO style, though of course it is ES6 only and requires some sort of transpilation to work backwards.

**Comment by benjamingr** [comment]
It's also possible to do:

    Object.assign(Object.create(Foo), {
           ..
    });

Although the version you have is shorter.

It's also worth mentioning that although it requires transpilation in theory because it is ES6 - all common browsers (IE, FF and Chrome and Safari) support this already - Chrome, Safari and FF for a while and IE since `__proto__` was added (IE11).

**Comment by jinbeomhong** [comment]
As you know, __proto__ is legacy feature.

**Comment by benjamingr** [comment]
@jinbeomhong that is not correct it was standardized in es2015

**Comment by jinbeomhong** [comment]
@benjamingr You are right :) But i'm not saying it was standardized but it is legacy feature.

[From ECMAScript 2015 specification](https://www.ecma-international.org/ecma-262/6.0/#sec-additional-ecmascript-features-for-web-browsers)

> This annex describes various legacy features and other characteristics of web browser based ECMAScript implementations...

> These features are not considered part of the core ECMAScript language. Programmers should not use or assume the existence of these features and behaviours when writing new ECMAScript code...

[From MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/proto)

> While `Object.prototype.__proto__` is supported today in most browsers, its existence and exact behavior has only been standardized in the ECMAScript 2015 specification as a legacy feature to ensure compatibility for web browsers. For better support, it is recommended that `Object.getPrototypeOf()` be used instead.
